### PR TITLE
Characterize and remove field driver convergence assertions

### PR DESCRIPTION
**Pull request by sethrj**
This demonstrates that #619 will be fixed by properly killing particles less than ~1 keV. The number of equation evaluations required to meet the FieldDriver 'accurate' tolerance increases rapidly with decreasing energy below 1 keV for electrons. At 1 MeV, ~10 steps are required to converge on a ~0.4cm step from a 10cm physics length. At 1 eV, ~250 are required for a 0.08cm step. So if we have low energy tracks in a low density material we're really screwed.

### CODE REVIEW

File: src/celeritas/field/FieldDriver.hh [hunk 1]
```diff
@@ -348,9 +342,6 @@ FieldDriver<StepperT>::one_good_step(real_type step, OdeState const& state) cons
         }
     } while (!succeeded && --remaining_steps > 0);
 
-    // TODO: loop check and handle rare cases if happen
-    CELER_ASSERT(succeeded);
```
Comment (whokion) [inline src/celeritas/field/FieldDriver.hh:352]: Removing these assertions (succeeded with an incomplete integration) cause problems in the upstream FieldPropagator.  So, propose to revert, but increase  the default max_nsteps from 100 to 500 for temporarily until all remaining problems of very low energy particle are understood and fixed.

Comment (sethrj) [reply src/celeritas/field/FieldDriver.hh:352]: Uh oh. What are the problems upstream?

Comment (whokion) [reply src/celeritas/field/FieldDriver.hh:352]: The first hit is at this [line](https://github.com/sethrj/celeritas/blob/619-field-driver/src/celeritas/field/FieldPropagator.hh#L231) and then cascade the problem to further at t[he end condition](https://github.com/sethrj/celeritas/blob/619-field-driver/src/celeritas/field/FieldPropagator.hh#L285).

**Comment by sethrj** [comment]
@whokion I think that between #648 and #654 the downstream assertions you noticed are no longer triggered. Since I'm no longer seeing any field-related assertions, can we merge this? Or would you like to try again on your side?

**Comment by whokion** [comment]
@sethrj please merge this now if urgent - I will test it later (this afternoon) and report if there are still remaining issues.

**Review by whokion — APPROVED** [review]

**Comment by sethrj** [comment]
Thanks. It's not too urgent: I've already been running stuff on a "test" branch that has all these things merged together.

**Comment by whokion** [comment]
@sethrj I confirm that demo-loop with cms2018 geometry run successfully without any assertion.  Superb job!

**Pull request merged**